bento/src/main/java/com/yelp/android/bento/core/ComponentGroup.java

```diff
@@ -412,6 +407,41 @@ public int findComponentOffset(@NonNull Component component) {
         return -1;
     }
 
+    /**
+     * Finds and returns the component at lowest level (leaf) that encompasses the index.
+     *
+     * @param index The index to search for.
+     * @return The lowest component in the tree.
+     */
+    @Override
+    public Component getLowestComponentAtIndex(int index) {
+        Component componentAtPos = componentAt(index);
+        Range range = rangeOf(componentAtPos);
+        if (range == null) {
+            throw new IllegalStateException("Found component, but no range associated.");
+        }
+
+        return componentAtPos.getLowestComponentAtIndex(index - range.mLower);
+    }
+
+    /**
+     * Similar to {@link #getLowestComponentAtIndex(int)}, but returns both the component and the
```

Comment:
If these methods do the same thing, couldn't we have `getLowestComponentAtIndex(int)` return `getLowestRangeValue(int index).mValue`?